include/seqan3/alignment/multiple/detail/align_multiple_seqan2_adaptation.hpp

```diff
@@ -0,0 +1,285 @@
+                seqan::appendValue(tmp, alphabet_type{seqan3::to_char(chr)});
+
+            seqan::appendValue(sequences, tmp);
+            seqan::appendValue(ids, dummy_name);
+            seqan::clear(tmp);
+        }
+
+        return std::make_pair(std::move(sequences), std::move(ids));
+    }
+
+    /*!\brief Create a vector of gapped SeqAn3 sequences from a SeqAn2 alignment graph.
+     * \tparam graph_type The type of the alignment graph resulting from a seqan::globalMsaAlignment.
+     * \param alignment_graph The alignment graph from a seqan::globalMsaAlignment.
+     * \return A std::vector of gapped SeqAn3 sequences representing the MSA result.
+     */
+    template <typename graph_type>
+    auto create_output(graph_type & alignment_graph)
+    {
+        using gapped_alphabet_type = seqan3::gapped<seqan3_alphabet_type>;
+
+        // convert the alignment graph into a string of all aligned sequences concatenated to each other
+        std::string concatenated_alignment_string;
+        seqan::convertAlignment(alignment_graph, concatenated_alignment_string);
+
+        // initialise the output vector of SeqAn3 sequences
+        auto & underlying_string_set = seqan::stringSet(alignment_graph);
+        std::vector<std::vector<gapped_alphabet_type>> output{seqan::length(underlying_string_set)};
+
+        size_t const alignment_length = concatenated_alignment_string.size() / output.size();
+
+        auto iter = output.begin();
+        for (auto && gapped_seq : concatenated_alignment_string | seqan3::views::char_to<gapped_alphabet_type>
+                                                                | seqan3::views::chunk(alignment_length))
+        {
+            iter->reserve(alignment_length);
+            std::ranges::copy(gapped_seq, std::cpp20::back_inserter(*iter++));
+        }
+
+        return output;
+    }
+
+private:
+    /*!\brief Predicate that represents whether `candidate_t` is a type that is not allowed as MSA configuration.
+     * \tparam candidate_t The type to test.
+     *
+     * \details
+     * The predicate's value is set to: NOT (band OR gap OR scoring).
+     */
+    template <typename candidate_t>
+    struct is_invalid_msa_config :
+        std::negation<std::disjunction<std::is_same<candidate_t, seqan3::align_cfg::band_fixed_size>,
+                                       is_type_specialisation_of<candidate_t, seqan3::align_cfg::gap>,
+                                       is_type_specialisation_of<candidate_t, seqan3::align_cfg::scoring_scheme>>>{};
+
+    /*!\brief Validate the given MSA configuration.
+     * \tparam configs_t The specified configuration elements.
+     */
+    template <typename ... configs_t>
+    void validate_configuration(seqan3::configuration<configs_t...> const &)
+    {
+        static_assert(seqan3::pack_traits::find_if<is_invalid_msa_config, configs_t...> == -1,
+                      "The given MSA configuration is not valid.");
+    }
```

Comment:
> And before we do this. Is that really necessary?

Well, we could also just ignore all other configuration types but I thought it might be good to know for the user which are allowed instead of just silently ignoring them :shrug: